net-p2p/retroshare/retroshare-0.6.7-r2.ebuild

```diff
@@ -0,0 +1,129 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit flag-o-matic desktop qmake-utils xdg
+
+DESCRIPTION="Friend to Friend secure communication and sharing application"
+HOMEPAGE="https://retroshare.cc"
+SRC_URI="https://download.opensuse.org/repositories/network:/retroshare/Debian_Testing/retroshare-common_${PV}.orig.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/RetroShare"
+# NOTE: GitHub releases/archive is impractical to build so we use the OBS repo
+# but they squash point releases which is bad for us
+
+LICENSE="AGPL-3 Apache-2.0 CC-BY-SA-4.0 GPL-2 GPL-3 LGPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="keyring cli +gui +jsonapi libupnp +miniupnp +service +sqlcipher plugins"
+
+REQUIRED_USE="
+	|| ( gui service )
+	?? ( libupnp miniupnp )
+	plugins? ( gui )
+	service? ( || ( cli jsonapi ) )
+"
+RDEPEND="
+	app-arch/bzip2
+	dev-libs/openssl:0=
+	sys-libs/zlib
+	keyring? ( app-crypt/libsecret )
+	gui? (
+		dev-qt/qtcore:5
+		dev-qt/qtgui:5
+		dev-qt/qtmultimedia:5
+		dev-qt/qtnetwork:5
+		dev-qt/qtprintsupport:5
+		dev-qt/qtxml:5
+		dev-qt/qtwidgets:5
+		dev-qt/qtx11extras:5
+		x11-libs/libX11
+		x11-libs/libXScrnSaver
+	)
+	libupnp? ( net-libs/libupnp:= )
+	miniupnp? ( net-libs/miniupnpc:= )
+	sqlcipher? ( dev-db/sqlcipher )
+	!sqlcipher? ( dev-db/sqlite:3 )
+	plugins? (
+		media-libs/speex
+		media-libs/speexdsp
+		<media-video/ffmpeg-5
+	)
+"
+# NOTE(setan): rapidjson might not be needed if not using jsonapi. This is to be tested
+DEPEND="${RDEPEND}
+	>=dev-libs/rapidjson-1.1.0
+	gui? ( dev-qt/designer:5 )
+"
+BDEPEND="
+	dev-build/cmake
+	dev-qt/qtcore:5
+	virtual/pkgconfig
+	gui? ( x11-base/xorg-proto )
+	jsonapi? ( app-text/doxygen )
+"
+PATCHES=( "${FILESDIR}/${P}-fix-miniupnp-api-v18.patch" )
+
+src_configure() {
+	local qconfigs=(
+		$(usex cli     '' 'no_')rs_service_terminal_login
+		$(usex keyring '' 'no_')rs_autologin
+		$(usex gui     '' 'no_')retroshare_gui
+		$(usex jsonapi '' 'no_')rs_jsonapi
+		$(usex service '' 'no_')retroshare_service
+		$(usex sqlcipher '' 'no_')sqlcipher
+		$(usex plugins '' 'no_')retroshare_plugins
+	)
+
+	local qupnplibs="none"
+	use miniupnp && qupnplibs="miniupnpc"
+	use libupnp && qupnplibs="upnp ixml"
+
+	# bug 907898
+	use elibc_musl && append-flags -D_LARGEFILE64_SOURCE
+
+	# REVIEW: qmake is deprecated
+	# https://github.com/RetroShare/RetroShare/tree/master/jsonapi-generator
+	eqmake5 CONFIG+="${qconfigs[*]}" \
+		RS_MAJOR_VERSION=$(ver_cut 1) \
+		RS_MINOR_VERSION=$(ver_cut 2) \
+		RS_MINI_VERSION=$(ver_cut 3) \
+		RS_EXTRA_VERSION="-gentoo-${PR}" \
+		RS_UPNP_LIB="${qupnplibs}"
+}
+
+src_install() {
+	use gui && dobin retroshare-gui/src/retroshare
+	use service && dobin retroshare-service/src/retroshare-service
+
+	insinto /usr/share/retroshare
+	doins libbitdht/src/bitdht/bdboot.txt
+	use gui && doins -r retroshare-gui/src/qss
+
+	dodoc README.asciidoc
+
+	if use gui; then
+		make_desktop_entry Retroshare
```

Comment:
Why is this changing?

```
git log -pC pull/37435
```

```diff
@@ -98,7 +103,7 @@ src_install() {
 	dodoc README.asciidoc
 
 	if use gui; then
-		make_desktop_entry retroshare
+		make_desktop_entry Retroshare
 
 		for i in 24 48 64 128 ; do
 			doicon -s ${i} "data/${i}x${i}/apps/retroshare.png"
```

First argument to make_desktop_entry is the binary name.

Reply:
If it comes to that, there's an upstream desktop file:

https://github.com/RetroShare/RetroShare/blob/master/data/retroshare.desktop

Why don't we use that directly? In fact, the upstream "install" routine should I think already handle that -- qmake does support installation. It might be possible to just use that directly.

Reply:
Using the upstream desktop file would be capitalized. The upstream desktop file uses:

```ini
Name=RetroShare
Exec=/usr/bin/retroshare %U
```

and the Name is capitalized. Your patch makes custom desktop file created by gentoo have the Exec be capitalized, which is very wrong.

Gentoo shouldn't be creating custom downstream desktop files without a good reason. Before 2013, upstream didn't provide one, which at the time meant the first version of retroshare added to Gentoo needed to create its own. But this has been a decade out of date, at this point.

Reply:
There is one in https://github.com/RetroShare/RetroShare/blob/master/data/retroshare.xpm and it is installed by the CMakeLists.txt to the directory the desktop file points at.

Although really, the desktop file format and icon theme standards suggest not hardcoding the filename...